### PR TITLE
DIS-32 make total credits bar unclickable

### PR DESCRIPTION
**Pull request by Byuljapcho**
**Description**
In the distributions branch on frontend, clicking on the total credits bar in the tracker caused the app to crash. So the solution is to make the total credits bar unclickable. 

**Implementation**
In the CourseBar.tsx, openCartPopup function is called when the CourseBar is clicked. I made the total credits bar unclickable by changing how the correct distribution was filtered from the distributions. 

**Testing**
Tested locally but only on a new plan with no courses added. I have two dev accounts, mkim224 and mkim25, but for some reason they don't have majors in the plans containing courses (and thus only the total credits bar showed up in the tracker). So I created a new plan which meant distribution bars were shown in the tracker, but I couldn't add new courses and check that course counting and distribution tracking are working after making the total credits bar unclickable. 

I would like to conduct more testing locally once I figure out the issues with my dev accounts.

### CODE REVIEW

**Comment by Byuljapcho** [comment]
Not sure why I still have prettier issues with these two files again...on my VSCode, they have no issues
[warn] lib/components/dashboard/menus/PlanEditMenu.tsx
[warn] lib/resources/commonTypes.tsx

File: lib/components/popups/CourseDisplayPopup.tsx [hunk 1]
```diff
@@ -168,46 +168,48 @@ const CourseDisplayPopup: FC = () => {
...
-      dispatch(updateSelectedPlan(newPlan));
-      const newPlanList = [...planList];
-      for (let i = 0; i < planList.length; i++) {
-        if (planList[i]._id === newPlan._id) {
-          newPlanList[i] = newPlan;
+  const handlePostAddCourse =
+    (plan: Plan) =>
+    (data): void => {
+      let newUserCourse: UserCourse;
+      if (data.errors === undefined && courseToShow !== null) {
```
Comment (mpark63) [inline lib/components/popups/CourseDisplayPopup.tsx:175]: ah i'm guessing a lot of these changes are from running prettier. Thanks miseok!

File: lib/components/dashboard/degree-info/CourseBar.tsx [hunk 2]
```diff
@@ -62,16 +57,12 @@ const CourseBar: FC<{
   const openCartPopup = async () => {
     dispatch(updateCartInvokedBySemester(false));
     // Filter for the correst distributions from redux store
-    let distr: UserDistribution = distribution;
-    // get fineReqs
-    if (distribution._id) {
-      distr = await getDistribution(distribution._id, token);
-    }
-    if (distr) {
```
Comment (mpark63) [inline lib/components/dashboard/degree-info/CourseBar.tsx:70]: You do need this code actually, to display fine requirements for selected distribution!

File: lib/components/dashboard/degree-info/CourseBar.tsx [hunk 3]
```diff
@@ -40,7 +36,6 @@ const CourseBar: FC<{
   const [plannedCredits, setPlannedCredits] = useState(distribution.planned);
   const [hovered, setHovered] = useState(false);
 
-  const token = useSelector(selectToken);
```
Comment (mpark63) [inline lib/components/dashboard/degree-info/CourseBar.tsx:43]: you need token to make the API call

File: lib/components/dashboard/degree-info/CourseBar.tsx [hunk 4]
```diff
@@ -62,16 +57,12 @@ const CourseBar: FC<{
...
     dispatch(updateCartInvokedBySemester(false));
     // Filter for the correst distributions from redux store
-    let distr: UserDistribution = distribution;
-    // get fineReqs
-    if (distribution._id) {
-      distr = await getDistribution(distribution._id, token);
-    }
-    if (distr) {
+    let distrs = distributions.filter((req) => req[0] === distribution.name)[0];
+    if (distrs) {
```
Comment (mpark63) [inline lib/components/dashboard/degree-info/CourseBar.tsx:61]: I think you should just make the total credits bar unclickable by adding a tailwind property to it; that would be in DistributionBarsJSX.tsx 

Screenshot 2023-03-02 at 2 50 07 PM

Comment (mpark63) [reply lib/components/dashboard/degree-info/CourseBar.tsx:61]: that being said, you probably need not touch this file.

**Review by mpark63 — CHANGES_REQUESTED** [review]
Hey Miseok, please see my comments! currently you made all the distribution bars unclickable. 

Also... it's largely my bad for not sharing the distributions database URI with you. Please connect to the one I share with you on slack, INSTEAD of the DEV DB.

Thanks and sorry again for the delay.

File: lib/components/dashboard/degree-info/CourseBar.tsx (hunk 3 above)
Comment (Byuljapcho) [reply lib/components/dashboard/degree-info/CourseBar.tsx:43]: sorry, added it back to the file

File: lib/components/dashboard/degree-info/CourseBar.tsx (hunk 2 above)
Comment (Byuljapcho) [reply lib/components/dashboard/degree-info/CourseBar.tsx:70]: reverted the file

File: lib/components/dashboard/degree-info/CourseBar.tsx (hunk 3 above)
Comment (Byuljapcho) [reply lib/components/dashboard/degree-info/CourseBar.tsx:43]: reverted file

**Pull request merged**